dipy/workflows/stats.py

```diff
@@ -248,37 +329,122 @@ def run(self, h5_files, no_disks=100, out_dir=''):
         for file_path in io_it:
 
             logging.info('Applying metric {0}'.format(file_path))
-            file_name = os.path.basename(file_path)[:-3]
-            df = pd.read_hdf(file_path)
 
-            if len(df) < 100:
-                raise ValueError("Dataset for Linear Mixed Model is too small")
+            file_name, bundle_name, save_name = self.get_metric_name(file_path)
+            print(" file name = ", file_name)
+            print("file path = ", file_path)
+
+            pvalues = np.zeros(no_disks)
+            warnings.filterwarnings("ignore")
+            # run mixed linear model for every disk
+            for i in range(no_disks):
+                disk_count = i+1
+                df = pd.read_hdf(file_path, where='disk=disk_count')
+
+                print("read the dataframe for disk number ", disk_count)
+                # check if data has significant data to perform LMM
+                if len(df) < 10:
+                    raise ValueError("Dataset for Linear Mixed Model is too small")
+
+                criteria = file_name + " ~ group"
+                md = smf.mixedlm(criteria, df,
+                                 groups=df["subject"])
+
+                mdf = md.fit()
 
-            all_bundles = df.bundle.unique()
-            # all_pvalues = []
-            for bundle in all_bundles:
-                sub_af = df[df['bundle'] == bundle]  # sub sample
-                pvalues = np.zeros(no_disks)
+                pvalues[i] = mdf.pvalues[1]
 
-                # run mixed linear model for every disk
-                for i in range(no_disks):
+            x = list(range(1, len(pvalues)+1))
+            y = -1*np.log10(pvalues)
+
+            save_file = os.path.join(out_dir, save_name + "_pvalues.npy")
+            np.save(save_file, pvalues)
+
+            save_file = os.path.join(out_dir, save_name + "_pvalues_log.npy")
+            np.save(save_file, y)
+
+            save_file = os.path.join(out_dir, save_name + ".png")
+            self.save_lmm_plot(save_file, file_name, bundle_name, x, y)
+
+
+class BundleShapeAnalysis(Workflow):
+    @classmethod
+    def get_short_name(cls):
+        return 'BS'
+
+    def run(self, subject_folder, threshold=6, out_dir=''):
+        """Workflow of bundle analytics.
+
+        Applies bundle shape similarity analysis on bundles of subjects and
+        saves the results in a directory specified by ``out_dir``.
+
+        Parameters
+        ----------
+
+        subject_folder : string
+            Path to the input subject folder. This path may contain
+            wildcards to process multiple inputs at once.
+
+        threshold : float (default 6)
```

Comment:
`(default 6)` appear twice and should be replaced by `, optional`

Reply:
changed